src/core/ext/transport/chttp2/transport/parsing.cc

```diff
@@ -92,10 +96,12 @@ grpc_error_handle grpc_chttp2_perform_read(grpc_chttp2_transport* t,
           return GRPC_ERROR_CREATE_FROM_CPP_STRING(absl::StrFormat(
               "Connect string mismatch: expected '%c' (%d) got '%c' (%d) "
               "at byte %d",
-              GRPC_CHTTP2_CLIENT_CONNECT_STRING[t->deframe_state],
+              get_utf8_safe_char(
```

Comment:
It's still valuable to keep this change because although we don't need to technically but having utf-8 compatible strings is worth it.

Reply:
The way this is being done means that the message will actually be corrupted if it's composed of non-UTF-8 characters.  If you just allow the non-UTF-8 characters here, then the change you're making in `StatusToProto()` will properly propagate the error message.  But with the change here, the error message will be corrupted, and people may not be able to read it.

Reply:
I don't understand why you think this message is corrupted. It shows both the code of character and character itself only when the character falls in the valid utf-8 range. like `A (65)`. For the case of invalid utf-8 code, it used to print like `Æ (198)' but it changes to ` (198)` by this change. It sill carries the information about the code 198 but it just doesn't print it as a character because it's not a utf-8 code.

Reply:
Oh, sorry, I was misreading this.  You're right, there is enough info here.